### PR TITLE
ddl: fix the compatibility of dropping non-exist column

### LINKED ISSUE #26137: DDL:1091 error message  is not compatible with mysql
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

MySQL
```sql
create table t (c int);
alter table t drop column not_exist_column;
# => ERROR 1091 (42000): Can't DROP 'not_exist_column'; check that column/key exists
```

TiDB

```sql
create table t (c int);
alter table t drop column not_exist_column;
# => error: mysql: 1091: column not_exist_column doesn't exist
```

### 2. What did you expect to see? (Required)
```
Can't DROP 'not_exist_column'; check that column/key exists
```
https://github.com/pingcap/tidb/blob/9cf25b6c24a340b0ee0c5c2d777abde90c6a8f9f/errors.toml#L84-L87
### 3. What did you see instead (Required)

```
error: mysql: 1091: column not_exist_column doesn't exist
```

### 4. What is your TiDB version? (Required)

```
 Release Version: v5.1.0                                          +
 Edition: Community                                               +
 Git Commit Hash: 8acd5c88471cb7b4d4c4a8ed73b4d53d6833f13e        +
 Git Branch: heads/refs/tags/v5.1.0                               +
 UTC Build Time: 2021-06-24 07:08:50                              +
 GoVersion: go1.16.4                                              +
 Race Enabled: false                                              +
 TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306+
 Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by sylzd**
### What problem does this PR solve?

Issue Number: close #26137  

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
Fix the compatibility error info of dropping non-exist column with mysql.
```

### CODE REVIEW

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -3413,3 +3413,13 @@ func (s *testIntegrationSuite3) TestTruncateLocalTemporaryTable(c *C) {
 	tk.MustExec("truncate table testt.t3")
 	tk.MustQuery("select * from testt.t3").Check(testkit.Rows())
 }
+
+func (s *testIntegrationSuite3) TestDropNoExitsColumn(c *C) {
```
Comment (wjhuang2016) [inline ddl/db_integration_test.go:3417]: This test is not needed, since the error message is covered by other tests.

Comment (sylzd) [reply ddl/db_integration_test.go]: That's right

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -3566,7 +3566,7 @@ func checkIsDroppableColumn(ctx sessionctx.Context, t table.Table, spec *ast.Alt
 	colName := spec.OldColumnName.Name
 	col := table.FindCol(t.VisibleCols(), colName.L)
 	if col == nil {
-		err = ErrCantDropFieldOrKey.GenWithStack("column %s doesn't exist", colName)
+		err = ErrCantDropFieldOrKey.GenWithStack("Can't DROP '%s'; check that column/key exists", colName)
```
Comment (xiongjiwei) [inline ddl/ddl_api.go:3569]: maybe `ErrCantDropFieldOrKey.GenWithStackByArgs(colName)` is ok?

Comment (sylzd) [reply ddl/ddl_api.go:3569]: Nope~ according to #26137, we should keep the same error message with the mysql

**Review by wjhuang2016 — APPROVED** [review]

File: ddl/ddl_api.go (hunk 2 above)
Comment (xiongjiwei) [reply ddl/ddl_api.go:3569]: right, but you can see the template of `ErrCantDropFieldOrKey`, it is `Can't DROP '%-.192s'; check that column/key exists`. No need to write the template again.

btw, there are some other usages of `ErrCantDropFieldOrKey`, could please check them?

Comment (sylzd) [reply ddl/ddl_api.go]: I see~Thanks to your suggestion, I modified another place as well.

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**